inc/dbutils.class.php

```diff
@@ -0,0 +1,903 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2017 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+
+/**
+ * Database utilities
+ *
+ * @since 9.2
+ */
+final class DbUtils {
+
+   /**
+    * Return foreign key field name for a table
+    *
+    * @param string $table table name
+    *
+    * @return string field name used for a foreign key to the parameter table
+    */
+   public function getForeignKeyFieldForTable($table) {
+      if (!Toolbox::startsWith($table, 'glpi_')) {
+         return "";
+      }
+      return str_replace("glpi_", "", $table)."_id";
+   }
+
+
+   /**
+    * Check if field is a foreign key field
+    *
+    * @param string $field field name
+    *
+    * @return boolean
+    */
+   public function isForeignKeyField($field) {
+      return preg_match("/._id$/", $field) || preg_match("/._id_/", $field);
+   }
+
+
+   /**
+    * Return table name for a given foreign key name
+    *
+    * @param string $fkname foreign key name
+    *
+    * @return string table name corresponding to a foreign key name
+    */
+   public function getTableNameForForeignKeyField($fkname) {
+      if (!$this->isForeignKeyField($fkname)) {
+         return '';
+      }
+
+      // If $fkname begin with _ strip it
+      if (Toolbox::startsWith($fkname, '_')) {
+         $fkname = substr($fkname, 1);
+      }
+
+      return "glpi_".preg_replace("/_id.*/", "", $fkname);
+   }
+
+   /**
+    * Return the plural of a string
+    *
+    * @param string $string input string
+    *
+    * @return string plural of the parameter string
+    */
+   public function getPlural($string) {
+      $rules = [
+         //'singular'         => 'plural'
+         'criterias$'         => 'criterias',// Special case (criterias) when getPlural is called on already plural form
```

Comment:
sic, don't kown where it's used, but later (not now) we should investigate why and replace the couple by [criterion->criteria]

Reply:
:D